subgraph-js/src/schema-helper/transformSchema.ts

```diff
@@ -112,9 +110,7 @@ export function transformSchema(
   function replaceType<T extends GraphQLType>(
     type: GraphQLNonNull<T>
   ): GraphQLNonNull<T>;
-  function replaceType(type: GraphQLNamedType): GraphQLNamedType;
-  function replaceType(type: GraphQLOutputType): GraphQLOutputType;
-  function replaceType(type: GraphQLInputType): GraphQLInputType;
+  function replaceType<T>(type: T): T;
```

Comment:
```suggestion
  function replaceType<T extends GraphQLType>(type: T): T;
```
I think we need a constraint on `T` here